javascript-carousel/main.js

```diff
@@ -0,0 +1,108 @@
+// DOM queries and other variables
+var imageList = document.querySelectorAll('img');
+var buttonsList = document.querySelectorAll('.button-image');
+var timerID = null;
+
+// image change every so and so seconds
+function imageChangeWithInterval() {
+  imageChangeForward();
+  timerID = setTimeout(imageChangeWithInterval, 1300);
+}
+imageChangeWithInterval();
+
+// image changes going forward
+function imageChangeForward() {
+  var currentImage = document.querySelector('.current-pokemon');
+  var currentButton = document.querySelector('.current-button');
+
+  var nextImage = currentImage.nextElementSibling;
+  var nextButton = currentButton.nextElementSibling;
+
+  if (currentImage.id === 'jigglypuff' || currentButton.id === 'lastbutton') {
+    nextImage = document.querySelector('#bulbasaur');
+    nextButton = document.querySelector('#firstbutton');
+  }
+
+  currentImage.classList.remove('current-pokemon');
+  currentImage.classList.add('hidden');
+  nextImage.classList.remove('hidden');
+  nextImage.classList.add('current-pokemon');
+
+  currentButton.classList.remove('current-button');
+  nextButton.classList.add('current-button');
+}
+
+// click right arrow moves image reel forward
+function clickRightChevron(event) {
+  var rightArrow = document.querySelector('.fa-chevron-right');
+
+  if (event.target !== rightArrow) {
+    return;
+  }
+  imageChangeForward();
+
+  clearTimeout(timerID);
+  timerID = setTimeout(imageChangeWithInterval, 1300);
+}
+document.addEventListener('click', clickRightChevron);
+
+// click left arrow moves image backward
+function clickLeftChevron(event) {
+  var leftArrow = document.querySelector('.fa-chevron-left');
+  var currentImage = document.querySelector('.current-pokemon');
+  var currentButton = document.querySelector('.current-button');
+
+  var previousImage = currentImage.previousElementSibling;
+  var previousButton = currentButton.previousElementSibling;
+
+  if (event.target !== leftArrow) {
+    return;
+  }
```

Comment:
Try to get guards out of the way immediately at the top of the function so you don't needlessly query the DOM. Try `Element.matches()`as mentioned before.